torchbenchmark/__init__.py

```diff
@@ -311,6 +311,42 @@ def check_example() -> None:
         else:
             module(*example_inputs)
 
+    @base_task.run_in_worker(scoped=True)
+    @staticmethod
+    def check_device() -> None:
+        instance = globals()["model"]
+
+         # Check this BenchmarkModel has a device attribute.
+        current_device = getattr(instance, 'device', None)
+        if current_device is None:
+            raise Exception('Missing device in BenchmarkModel.')
+
+        model, inputs = instance.get_module()
+        model_name = model.name if hasattr(model, 'name') else None
+
+        # Check the model tensors are assigned to the expected device.
+        for t in list(model.parameters()):
+            model_device = t.device.type
+            if model_device != current_device:
+                raise Exception(f'Model {model_name} was not set to the' \
```

Comment:
Same as above. Probably `RuntimeError`? (That's what we raise for when you try to call an op with incompatible arguments.)